### ISSUE TITLE
"only" seems to throw an error when used with <c-component is="" />

### ISSUE THREAD

**Issue by alorence** [wrabit/django-cotton#196]
I have an alert component

`cotton/alert.html`
```html
<c-vars class/>
<div role="alert" class="alert{% if class %} {{ class }}{% endif %}" {{ attrs }}>
  <c-icon.info />  {{ slot }}
</div>
```

I can use it that way

`page.html`
```html
<c-alert class="margin-bottom-6">Important message</c-alert>
```

And finally, if I don't want class 'margin-bottom-6' to be passed to icon, I use 'only' attribute
```html
<c-vars class/>
<div role="alert" class="alert{% if class %} {{ class }}{% endif %}" {{ attrs }}>
  <c-icon.info only />{{ slot }}
</div>
```

But now, I want to make icon customizable:
```html
<c-vars class/>
<div role="alert" class="alert{% if class %} {{ class }}{% endif %}" {{ attrs }}>
  {% if icon %}
    <c-component is="icon.{{ icon }}" only />
  {% endif %}
  {{ slot }}
</div>
```

Unfortunately, this version throws the error:

> AttributeError: 'NoneType' object has no attribute 'engine'

Removing the 'only' attribute solve the error, but the class is passed down to icon component, which is not the desired behavior.

Full traceback:

```
Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 880, in _resolve_lookup
    current = current[bit]
              ^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/context.py", line 83, in __getitem__
    raise KeyError(key)
KeyError: 'icon'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 886, in _resolve_lookup
    if isinstance(current, BaseContext) and getattr(
                                            ^^^^^^^^
AttributeError: type object 'Context' has no attribute 'icon'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 896, in _resolve_lookup
    current = current[int(bit)]
                      ^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'icon'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 715, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 847, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 903, in _resolve_lookup
    raise VariableDoesNotExist(
django.template.base.VariableDoesNotExist: Failed lookup for key [icon] in [{'True': True, 'False': False, 'None': None}, {'cotton_data': {'stack': [{'key': 'component', 'attrs': <django_cotton.templatetags.Attrs object at 0x7fca95eac200>, 'slots': {}}], 'vars': {}}}]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 720, in resolve
    string_if_invalid = context.template.engine.string_if_invalid
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'engine'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django_cotton/templatetags/_attr.py", line 32, in render
    content = self.nodelist.render(context)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 968, in render_annotated
    if context.template.engine.debug:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'engine'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/exception.py", line 57, in inner
    response = response_for_exception(request, exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/exception.py", line 140, in response_for_exception
    response = handle_uncaught_exception(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/exception.py", line 181, in handle_uncaught_exception
    return debug.technical_500_response(request, *exc_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django_extensions/management/technical_response.py", line 40, in null_technical_500_response
    raise exc_value.with_traceback(tb)
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/core/handlers/base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/defaulttags.py", line 321, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django_cotton/templatetags/_component.py", line 69, in render
    output = template.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 177, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django_cotton/templatetags/_vars.py", line 48, in render
    output = self.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/defaulttags.py", line 321, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django_cotton/templatetags/_component.py", line 115, in render
    return super().render(isolated_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django_cotton/templatetags/_component.py", line 52, in render
    default_slot = self.nodelist.render(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.cache/pypoetry/virtualenvs/myproject-4kgUsWzL-py3.12/lib/python3.12/site-packages/django/template/base.py", line 968, in render_annotated
    if context.template.engine.debug:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'engine'
```

**Comment by mattbha** [comment]
I can take poke around at this

**Comment by mattbha** [comment]
high level, here's how I'd write the customizable icon. i drop the if (and accept the trailing space) and make sure to define all consumed vars in c-vars if you dont want it going to attrs.

```
<c-vars class icon />
<div role="alert" class="alert {{ class }}" {{ attrs }}>
  {% if icon %}
    <c-component is="icon.{{ icon }}" only />
  {% endif %}
  {{ slot }}
</div>
```

**Comment by wrabit** [comment]
@alorence released a fix in [1.2.0](https://github.com/wrabit/django-cotton/releases/tag/v1.2.0), please let me if the issue still persists! Thanks. Cheers @mattbha, I had already started the poking around earlier on!

**Comment by alorence** [comment]
@mattbha Indeed I forgot to add 'icon' attribute to c-vars, but this is unrelated with the reported bug.

@wrabit Thanks a lot, I can confirm #197 and 1.2.0 release fixed the error when using 'only' attribute in dynamic component.

Best regards
